### DISCUSSION TITLE
Feature suggestion: handling two tokens as a way around rate limits

### DISCUSSION THREAD

**Discussion by SachaG** [plhery/node-twitter-api-v2#83] [Ideas]
I was asking about rate limits on the Twitter dev forums and someone suggested this:

> you should use both the App auth (bearer token) and user auth (access token) together to give you 30 calls in 15 minutes

Would it be feasible to instantiate a client with both kinds of tokens and have this library automatically switch from one to the other when rate limits are being hit?

**Comment by alkihis** [comment]
Hello,
No, you can't instanciate a client with different kind of auth tokens.

The client can't decide which token should be used for a single request, and this can create situations where a single client returns different results depending on when it's used (app-only tokens and client tokens sometimes leads to different payload results). 

**Plus**, deciding those kind of things depending on rate limit cached information is not a good idea, because *by nature*, app-only token shares rate limits with every client associated with the application (so, possible other machines using the token).
